server/src/routes/routes.ts

```diff
@@ -86,4 +87,102 @@ router.post('/auth/login', async (req: Request, res: Response) => {
   res.json({ found });
 });
 
-export { router };
\ No newline at end of file
+
+router.post('/api/saveFile', (req : Request, res: Response) => {
+  const { username, workspace, filename, fileData } = req.body;
+  const path = './user-files/' + username + '/workspaces/' + workspace + '/' + filename;
+
+  try {
+    writeFileSync(path, fileData);
+    // save to volume here
+    res.json({ path: path });
+  } catch (err) {
+    res.json({ error: err });
+  }
+});
+
+router.post('/api/updateFile', (req : Request, res: Response) => {
+  const { username, workspace, filename, fileData } = req.body;
+  const path = './user-files/' + username + '/workspaces/' + workspace + '/' + filename;
+  if (!existsSync(path)) {
+    res.json({ error: 'Invaild path' });
+    return;
+  }
```

Comment:
Avoid sync functions (`existsSync`, `writeFileSync`) as they are synchonous functions + all execution in the server is blocked. 
Use async-await instead.